bip-cc.md

```diff
@@ -0,0 +1,207 @@
+[bip-0016][BIP16 specs]: a `CHECKSIG` operation accounts for `1` and a `CHECKMULTISIG` accounts for
+the number of public keys associated, or `20` if the number of public keys is greater than `16`. A
+`CHECKMULTISIG` not directly preceded by a minimally-pushed number between `1` and `16` (included)
+accounts for `20`.
+
+Transactions whose (witness-stripped) serialized size is exactly 64 bytes are made invalid.
+
+The coinbase transaction's `nLockTime` field must be set to the height of the block minus `1`[^2]
+and its `nSequence` field must not be equal to `0xffffffff`.
+
+# Rationale
+
+The restrictions on the timestamp of the first and last blocks of a difficulty adjustment period fix
+the timewarp and Murch-Zawy vulnerabilities[^3]. The latter poses mostly theoretical concerns but is
+extremely low risk to fix: the duration of an adjustment period has never been, and should never be,
+negative. The former is fixed by preventing the timestamp of the first block of a difficulty period
+to be lower than the previous block's, with a two hours grace period. A [previous
+proposal][BIP-XXXX] to fix timewarp used a ten minutes grace period instead, also adopted for
+[testnet4][BIP94 timewarp]. Out of an abundance of caution and because it only trivially worsens the
+block rate increase under attack, a two hours grace period is used here[^4].
+
+Disabling some Script operations and functionalities was [previously proposed][BIP-XXXX] to reduce
+the worst case block validation time but was met with resistance due to confiscation concerns[^5]. A
+delicate balance needs to be struck between minimizing the confiscation risks of a mitigation, even
+if merely theoretical, and bounding the costs one could impose on all other users of the system. To
+this effect a limit on the number of potentially executed signature operations pinpoints exactly the
+harmful behaviour, leaving maximum flexibility in how Script functionalities may have been used.
+Such a limit reduces the worst case block validation time by a factor of 40 and drastically
+increases the preparation cost of an attack to make it uneconomical for a miner[^6]. The `2500`
+value was chosen as the tightest value that did not make any non-pathological standard transaction
+invalid[^7].
+
+In the presence of 64 bytes transactions a block header's Merkle root may be valid for different
+sets of transactions. This is because in the Merkle tree construction a 64 bytes transaction may be
+interpreted as the catenation of two 32 bytes hashes, or the catenation of two 32 bytes hashes may
+be interpreted as a transaction. The former allows to fake a block inclusion proof and the latter
+makes it such that for a valid block the Merkle root in the block header is not a unique identifier
+for the corresponding list of valid transactions[^8]. 64 bytes transactions can only contain an
+output Script that lets anyone spend the funds, or burns them.  They have also been non-standard for
```

Comment:
```suggestion
output script that lets anyone spend the funds, or burns them.  They have also been non-standard for
```

Reply:
Rationale?

Reply:
"Script" is the name of the scripting language, while a "script" is any program written in Script. See [this handy diagram](https://bitcoin.stackexchange.com/a/122808/121614) I've made, and for even more details check out [Murch's transaction terminology BIP](https://github.com/murchandamus/bips/blob/730a813124680a417f44bfe8117afbc89fd1ea6c/bip-tx-terminology.mediawiki).